### PR TITLE
chore: pass headers to request

### PR DESCRIPTION
**Pull request by ashleythedeveloper**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [x] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

This PR adds the ability to pass custom headers to requests across the tier 1 test suite. This update allows users to specify headers such as authorization tokens in the configuration, making the test suites more flexible when interacting with services that require specific HTTP headers.

It also includes relevant documentation updates and test cases to validate this new feature, ensuring headers are plain objects and that they are passed correctly in requests.

**NOTE:** It has been uncovered that the render tests are failing. This was hidden due to the authorisation error this PR seeks to resolve. Another PR will be raised to fix the tests.

## Mobile & Desktop Screenshots/Recordings
Screenshot 2024-10-22 at 10 03 14 PM
Screenshot 2024-10-22 at 10 03 18 PM
Screenshot 2024-10-22 at 10 03 25 PM

## Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📕 storybook
- [x] 🦖 Documentation Site
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: packages/vc-test-suite/httpService.ts [hunk 1]
```diff
@@ -9,6 +9,13 @@ export const request = async (params: AxiosRequestConfig) => {
     headers: defaultHeaders,
   });
 
+  // Check if user defined headers is an object and contains only string values
+  if (params.headers) {
```
Comment (ldhyen99) [inline packages/vc-test-suite/httpService.ts:13]: Hi, should we move this check before we call `axios.create`? If params.headers already contains conflicting values (like another Content-Type), they will not merge properly. The axios.create() method will only use defaultHeaders unless manually merged.

File: packages/vc-test-suite/__tests__/httpService.test.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    await expect(request({ headers: [] as any })).rejects.toThrow(
+      'Headers specified in the config must be a plain object with string values.',
+    );
+  });
+
+  it('should throw an error if headers contain non-string values', async () => {
+    await expect(request({ headers: { test: 123 } as any })).rejects.toThrow(
+      'Headers specified in the config must be a plain object with string values.',
+    );
+  });
```
Comment (huynguyen-hl) [inline packages/vc-test-suite/__tests__/httpService.test.ts:31]: I think we should check whether `instance.request` has been called and also assert that the correct parameters have been passed into `instance.request(params)`.

File: packages/vc-test-suite/__tests__/httpService.test.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import { request } from '../httpService';
+import axios from 'axios';
+
+jest.mock('axios');
+
+const mockedAxios = axios as jest.Mocked<typeof axios>;
+
+describe('httpService', () => {
```
Comment (huynguyen-hl) [inline packages/vc-test-suite/__tests__/httpService.test.ts:8]: Should we add an additional test case for when the `request` function is executed without the `headers` in the `params`?

File: packages/vc-test-suite/tests/RenderTemplate2024/render-template-2024.test.ts [hunk 4]
```diff
@@ -98,34 +101,31 @@ describe('RenderTemplate2024', function () {
...
-        documents: [
-          {
-            type: 'RenderTemplate2024',
-            renderedTemplate: 'PHA+SmFuZSBEb2U8L3A+',
-          },
-        ],
-      });
-    },
-  );
+  reportRow("should successfully render when 'mediaQuery' is not provided", config.implementationName, async () => {
```
Comment (huynguyen-hl) [inline packages/vc-test-suite/tests/RenderTemplate2024/render-template-2024.test.ts:104]: I think we can update this test case title to `should render successfully when ‘mediaQuery’ is not provided` and then remove the code comment `// TODO: check grammar`.

File: packages/vc-test-suite/httpService.ts (hunk 1 above)
Comment (ashleythedeveloper) [reply packages/vc-test-suite/httpService.ts:9]: Good catch

**Review by ldhyen99 — DISMISSED** [review]

**Review by namhoang1604 — APPROVED** [review]

**Review by huynguyen-hl — APPROVED** [review]

File: packages/vc-test-suite/httpService.ts [hunk 5]
```diff
@@ -1,12 +1,22 @@
 import axios, { AxiosRequestConfig } from 'axios';
+import _ from 'lodash';
```
Comment (ldhyen99) [inline packages/vc-test-suite/httpService.ts:2]: Hi, maybe we forgot to install this library?
image

image

**Review by namhoang1604 — APPROVED** [review]

**Review by huynguyen-hl — APPROVED** [review]

**Review by ldhyen99 — APPROVED** [review]

**Pull request merged**